contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,161 @@
+> How is iteration like in ReactJs:
+>
+```js
+// ReactJs (jsx)
+var Iteration = React.createClass({
+   getInitialState() {
+     return {
+       array: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+     }
+   },
+   render() {
+     this.state.array.map(function(date) {
+       return (
+         <span>{date}</span>
+       )
+     });
+   }
+});
+ReactDOM.render(<Iteration />, document.getElementById('array'));
+```
+```html
+<!-- ReactJs (html) -->
+<div id="array"></div>
+```
+>  How is iteration like in VueJs:
+>
+```js
+// VueJs (js)
+var Iteration = new Vue({
+  el: '#array',
+  data: {
+    array: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+  }
+});
+```
+```html
+<!-- VueJs (html) -->
+<div id="array">
+  <span v-for="date in array">{{date}}</span>
+</div>
+```
+
+2. <b>Progressive:</b>
+VueJs is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you have a large application, you can plug VueJs into just a part of your application without disturbing the other components. A quote from Evan You - the founder of VueJs is as follow:
+> Vue.js is a more flexible, less opinionated solution (than Angular). That allows you to structure your app the way you want it to be, instead of being forced to do everything the Angular way (Angular requires a certain way to structure an application, making it hard to introduce Angular into an already built project). It’s only an interface layer so you can use it as a light feature in pages instead of a full blown SPA (single-page application).
+>
+> -- <cite>Evan You</cite> [[source]](https://www.valuecoders.com/blog/technology-and-apps/vue-js-comparison-angular-react/)
+
+3. <b>Versatile:</b>
+VueJs is perfectly capable of powering sophisticated single-page applications when used in combination with modern tooling and supporting libraries.
+
+4. <b>Clean:</b>
+VueJs syntax is simple and this can make the HTML pages very clean. This would allow user interfaces built by VueJs to be more maintainable and testable.
+
+## Drawbacks of VueJS
+
+1. <b>Relatively small size community:</b>
+VueJs is a relatively new JavaScript framework as compared to Angular and React. The size of the community for VueJs is therefore relatively small. Although small size community means you can differentiate yourself from other JavaScript developers, it also means there are fewer resources such as tutorials and problem-shooting guides.
+
+2. <b>Language barriers:</b>
+A majority of users of VueJs are the Chinese as VueJs is developed by a Chinese American. He is supportive of the Chinese community and hence a lot of the existing plugins are written in Chinese. There might be some language barriers for an English speaking developer seeking for VueJs resources.
+
+## Other Advantages and Drawbacks of VueJs
```

Comment:
This is not worthy of a heading. In any case, these articles are not about 'Other' advantages and disadvantage. They are about _all_ advantages and disadvantages. That's why I suggested to merge the advantage and disadvantages into one section so that you can avoid this awkward heading. By the way, use one of these pairs pros-cons, benefits-drawbacks, advantages-disadvantages